### ISSUE TITLE
[HOLD App/pull/15204] [$1000] Settings -Text is not aligned In Contact method page

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#15581]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Log in any account
2. Go to Settings -Profile
3. Tap on Contact method

## Expected Result:
Text is aligned In Contact method page

## Actual Result:
Text is not aligned In Contact method page

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1,2,77,1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![contact](https://user-images.githubusercontent.com/93399543/223608490-917fbc9a-de86-41f6-96da-0d1fbd03a126.png)

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013e1a472f9df5fd6c</li>
        <li>Upwork Job ID: 1634164425334525952</li>
        <li>Last Price Increase: 2023-03-10</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mateocole (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@mateocole Eep! 4 days overdue now. Issues have feelings too...

**Comment by MelvinBot** [comment]
@mateocole Eep! 4 days overdue now. Issues have feelings too...

**Comment by mateocole** [comment]
Looks a bit misaligned, going to have engineering take a look and see if this is indeed an issue and if design or engineering needs to handle

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jasperhuangg (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by jasperhuangg** [comment]
@kbecciv I'm unable to see the image provided

<img width="544" alt="Screenshot 2023-03-07 at 3 55 02 PM" src="https://user-images.githubusercontent.com/31285285/223582480-7d4441cd-b583-4028-a70e-eb99b590402c.png">

**Comment by mountiny** [comment]
Calls tack could take this one

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~013e1a472f9df5fd6c

**Comment by MelvinBot** [comment]
📣 @narefyev91 You have been assigned to this job by @mountiny!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by MelvinBot** [comment]
Current assignee @mateocole is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mollfpr (`External`)

**Comment by MelvinBot** [comment]
Current assignee @jasperhuangg is eligible for the External assigner, not assigning anyone new.

**Comment by PrashantMangukiya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Settings >> Profile >> Contact Methods screen, when it show email address with `Resend` button (if email not verified) , or Phone number with `Resend` button.  So if any contact method not verified, it will render  section with `Resend` button and other section render as normal info. So here section that render with `Resend` button is not aligned with other section render as normal.

### What is the root cause of that problem?
We are rendering normal info i.e. email or phone (if verified) via code below:
https://github.com/Expensify/App/blob/bff8b0bda5edf9f7c4bc52bb76a8bfa482053ca7/src/pages/settings/Profile/Contacts/LoginField.js#L93-L102

We are rendering email and phone (if not verified) with `Resend` button via code below.
https://github.com/Expensify/App/blob/bff8b0bda5edf9f7c4bc52bb76a8bfa482053ca7/src/pages/settings/Profile/Contacts/LoginField.js#L104-L125

So there is not any negative margin applied to `<View style={[styles.mt2]}>` at line 104 that render view with `Resend` button. So it will not aligned with other content rendered via `<View style={[styles.mln8, styles.mrn8]}>` that result to left align mismatch when any contact method not verified (i.e. email or phone vice versa).

### What changes do you think we should make in order to solve the problem?
We have to adjust negative margin for view with 'Resend' button so it aligned with other view. So at line 104 we have to add style with left negative margin as shown below. (No need to set right negative margin because Resend button aligned perfect at right at present. We can also adjust similar way if need.)
```javascript
<View style={[styles.mt2, styles.mln3]}>
 ...
</view>
```
and, within `/src/styles/utilities/spacing.js` file we have to create style similar to this
https://github.com/Expensify/App/blob/c0f14f444a62fe8d0f70425f36251d2f962496ba/src/styles/utilities/spacing.js#L144-L146

i.e. Add new style `mln3` like this.
```javascript
  mln3: {
      marginLeft: -12,
  },
```
So this way we can adjust negative margin and show correct alignment.

### What alternative solutions did you explore? (Optional)
None

**Comment by mountiny** [comment]
Lets leave this to the community, @mollfpr can you please review the proposals coming in

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Text is not aligned on the same level in the contact method page

### What is the root cause of that problem?
Paddings and margins are not applied correctly for the screen

### What changes do you think we should make in order to solve the problem?
https://github.com/Expensify/App/blob/657b439007004a5a128fc76df00327cb14203915/src/pages/settings/Profile/Contacts/LoginField.js#L90
Remove the horizontal padding of screen from above and negative margin from below snippet. 
https://github.com/Expensify/App/blob/657b439007004a5a128fc76df00327cb14203915/src/pages/settings/Profile/Contacts/LoginField.js#L93
MenuItem has the correct padding sizes with `paddingHorizontal: 20, paddingVertical: 12` and same horizontal padding(`styles.ph5`) we can apply at below 2 places which will align the content correctly.

https://github.com/Expensify/App/blob/657b439007004a5a128fc76df00327cb14203915/src/pages/settings/Profile/Contacts/LoginField.js#L104
https://github.com/Expensify/App/blob/657b439007004a5a128fc76df00327cb14203915/src/pages/settings/Profile/Contacts/LoginField.js#L129

**Comment by MelvinBot** [comment]
📣 @PrashantMangukiya You have been assigned to this job by @jasperhuangg!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by jasperhuangg** [comment]
Ahh sorry @PrashantMangukiya based on how the LoginField will soon be deprecated, let's put this issue on HOLD and see if it's still worth fixing later on. Sorry for the confusion.

**Comment by jasperhuangg** [comment]
This issue may be invalidated by https://github.com/Expensify/App/pull/15204

**Comment by PrashantMangukiya** [comment]
@jasperhuangg  Thanks for the updates.

**Comment by mollfpr** [comment]
@Beamanator Can you confirm that we don't need any for `LoginField` at the moment?

**Comment by Beamanator** [comment]
Yes @mollfpr  thanks for the tag - @cristipaval is working on the "Add new contact method" PR here: https://github.com/Expensify/App/pull/15330 (still draft for now)

**Comment by mateocole** [comment]
moving to weekly since we are waiting on the other PR's

**Comment by mateocole** [comment]
Going to be OOO, reassigning bug

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by CortneyOfstad** [comment]
I'm heading OoO as well so reassigning so this is not delayed.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by CortneyOfstad** [comment]
Let's try this again

**Comment by MelvinBot** [comment]
Triggered auto assignment to @JmillsExpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by jasperhuangg** [comment]
We're still on HOLD for https://github.com/Expensify/App/pull/15330, not overdue. Demoting priority to Weekly so it stops going overdue

**Comment by jasperhuangg** [comment]
https://github.com/Expensify/App/pull/15330 is on staging! Closing this out since the page looks completely different now, and this issue doesn't manifest:

<img width="379" alt="Screenshot 2023-04-12 at 1 05 11 PM" src="https://user-images.githubusercontent.com/31285285/231572069-7ba9c148-e075-4f14-ae87-4706eb66a786.png">
